animation_nodes/nodes/material/object_material_output.py

```diff
@@ -23,13 +32,40 @@ def getExecutionFunctionName(self):
 
     def executeSingle(self, object, material):
         if object is None or not hasattr(object.data, "materials"): return object
-        object.data.materials.clear()
-        object.data.materials.append(material)
+
+        objectMaterials = object.data.materials
+        if not self.appendMaterials: objectMaterials.clear()
+
+        if object.type == "GPENCIL":
+            if material is not None:
+                if not material.is_grease_pencil:
+                    bpy.data.materials.create_gpencil_data(material)
+                self.setObjectMaterial(objectMaterials, material)
+            return object
```

Comment:
Don't make a distinction here. `create_gpencil_data` shouldn't be called, it is only supposed to set materials, not edit them.

Reply:
Omar, there is a difference between the _material_ for the _Mesh or Curve_ type object and material for the _GPencil_ object. We can't assign normal material (means material of the _Mesh_ or _Curve_) to GPencil object, we have to first convert that material into the gpencil-material.

Reply:
We can assign normal materials to GP objects. Yes, it won't be very useful, but that's what the user asked for. Ensuring that the material has GP data is the user's responsibility.

Reply:
Okay.